### DISCUSSION TITLE
Mapping Dynamic to specified object, Boolean property doesn't get mapped

### DISCUSSION THREAD

**Discussion by cammurray** [LuckyPennySoftware/AutoMapper#3684] [NA]
### Source/destination types

```csharp

// typed object
public class Foo
{
public string name { get; set; }
public Boolean valid { get; set; }
}

//someDynamicObject is coming from an API - Debug shows 'valid' property set to True in DynamicObject, can replicate with values True, "True" and 1 (int).

Foo FooBar = mapper.map<Foo>(someDynamicObject);

```

### Mapping configuration

```csharp
            var configuration = new MapperConfiguration(cfg => {});
            var mapper = configuration.CreateMapper();
```

### Expected behavior

Foo.valid = true (as per incoming DynamicObject)

### Actual behavior

Foo.valid = false

All other properties, e.g the name (string) are correctly mapped.

**Comment by cammurray** [comment]
On further look, this might not be a bug but my misunderstanding that you need to create a mapper for string to boolean. Will close off for now..

**Comment by cammurray** [comment]
Ok, so even with mapping configuration, valid is showing false even though the Dynamic has a value of True for that property.

I've tried the above with this mapping configuration and get the same result:

```csharp
            var configuration = new MapperConfiguration(cfg => {
                cfg.CreateMap<string, bool>().ConvertUsing(str => str.ToUpper() == "TRUE");
            });
            var mapper = configuration.CreateMapper();
```

**Reply by lbargaoanu** [reply]
It works for me. Add a failing test [here](https://github.com/AutoMapper/AutoMapper/blob/master/src/UnitTests/Mappers/DynamicMapperTests.cs).

**Reply by cammurray** [reply]
Hi @lbargaoanu 

Is it working for you using a boolean type? 

In the bug i commented that it works for strings, and other classes, but doesn't appear to be working when mapping from dynamic to bool. Checked the unit test you commented as well, and it doesnt appear that the unit test is testing anything other than strings?

Thanks for checking.

**Reply by lbargaoanu** [reply]
As I've already said, it works for me.
